rp2040-hal/src/multicore.rs

```diff
@@ -140,13 +140,36 @@ impl<'p> Core<'p> {
         }
     }
 
-    fn inner_spawn(
-        &mut self,
-        wrapper: *mut (),
-        entry: *mut (),
-        stack: &'static mut [usize],
-    ) -> Result<(), Error> {
-        if let Some((psm, ppb, sio)) = self.inner.as_mut() {
+    /// Spawn a function on this core.
+    pub fn spawn<F>(&mut self, stack: &'static mut [usize], entry: F) -> Result<(), Error>
+    where
+        F: FnOnce() -> bad::Never + Send + 'static,
+    {
+        if let Some((psm, ppb, fifo)) = self.inner.as_mut() {
+            // The first two ignored `u64` parameters are there to take up all of the registers,
+            // which means that the rest of the arguments are taken from the stack,
+            // where we're able to put them from core 0.
```

Comment:
Still a hack, but better than the MULTICORE_TRAMPOLINE used before. :+1:

Reply:
Agreed! My first impression was that it was too clever, but I've warmed up to it.